server_environment/models/server_env_mixin.py

```diff
@@ -0,0 +1,384 @@
+# Copyright 2018 Camptocamp (https://www.camptocamp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from functools import partialmethod
+
+from lxml import etree
+
+from odoo import api, fields, models
+from ..serv_config import serv_config
+
+_logger = logging.getLogger(__name__)
+
+
+class ServerEnvMixin(models.AbstractModel):
+    """Mixin to add server environment in existing models
+
+    Usage
+    -----
+
+    ::
+
+        class StorageBackend(models.Model):
+            _name = "storage.backend"
+            _inherit = ["storage.backend", "server.env.mixin"]
+
+            @property
+            def _server_env_fields(self):
+                return {"directory_path": {'getter': 'get'}}
+
+    With the snippet above, the "storage.backend" model now uses a server
+    environment configuration for the field ``directory_path``.
+
+    Under the hood, this mixin automatically replaces the original field
+    by an env-computed field that reads from the configuration files.
+
+    By default, it looks for the configuration in a section named
+    ``[model_name.Record Name]`` where ``model_name`` is the ``_name`` of the
+    model with ``.`` replaced by ``_``. Then in a global section which is only
+    the name of the model. They can be customized by overriding the method
+    :meth:`~_server_env_section_name` and
+    :meth:`~_server_env_global_section_name`.
+
+    For each field transformed to an env-computed field, a companion field
+    ``<field>_env_default`` is automatically created. When it's value is set
+    and the configuration files do not contain a key, the env-computed field
```

Comment:
`a key` seems a bit too generic.

You might want to state `a matching key` or `a key for that field`

Reply:
done